barretenberg/cpp/src/barretenberg/honk/transcript/transcript.hpp

```diff
@@ -79,24 +79,35 @@ template <typename FF> class BaseTranscript {
     TranscriptManifest manifest;
 
     /**
-     * @brief Compute c_next = H( Compress(c_prev || round_buffer) )
-     *
+     * @brief Compute next challenge c_next = H( Compress(c_prev || round_buffer) )
+     * @details This function computes a new challenge for the current round using the previous challenge
+     *          and the current round data, if they are exist.
+     *          It clears the current_round_data if nonempty after computing the challenge to minimize how much we
+     *          compress. It also sets previous_challenge_buffer to the current challenge buffer to set up next function
+     * call.
      * @return std::array<uint8_t, HASH_OUTPUT_SIZE>
      */
-    [[nodiscard]] std::array<uint8_t, HASH_OUTPUT_SIZE> get_next_challenge_buffer() const
+    [[nodiscard]] std::array<uint8_t, HASH_OUTPUT_SIZE> get_next_challenge_buffer()
     {
-        // Prevent challenge generation if nothing was sent by the prover.
-        ASSERT(!current_round_data.empty());
-
-        // concatenate the hash of the previous round (if not the first round) with the current round data.
+        // Empty buffer to compare against
+        std::array<uint8_t, HASH_OUTPUT_SIZE> empty_challenge_buffer{};
```

Comment:
Is this only needed to check if there was a previous challenge or not? It seems clearer to just use Adrians original `if (round_number > 0)` check. Maybe I'm missing something? `if (!no_previous_challenge)` is kind of icky :)

Reply:
can't use round_number because in the 0th round, we might want to call this function multiple times. I could change the bool to first_challenge instead to avoid double negative?

Reply:
I see. Maybe just a `first_challenge` bool that's initialized to true then gets set to false. I just find it a bit hard to understand what the intent is as you have it

Reply:
good idea, just did this instead